src/types.ts

```diff
@@ -1,26 +1,90 @@
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { ethers } from 'ethers';
+
+import { SupportedChainId } from './config';
 import { Asset, Transaction } from './gnosis-safe/types';
-import { Proposal, TokenMetaData, Vote, zDAO } from './snapshot-io/types';
+import {
+  Proposal,
+  ProposalDetail,
+  ProposalResult,
+  TokenMetaData,
+  Vote,
+} from './snapshot-io/types';
 
 export interface SnapshotConfig {
+  // uri to Snaphost Hub
   serviceUri: string;
-  ipfsUri: string;
+  // ipfs gateway
+  ipfsGateway: string;
 }
 
 export interface GnosisSafeConfig {
+  // uri to Gnosis Safe service
   serviceUri: string;
-  safeAddress: string;
+  // uri to Gnosis Safe gateway
   gateway: string;
 }
 
+export interface zNAConfig {
+  // address to zDAOCore contract
+  contract: string;
+  // web3 provider
+  provider: ethers.providers.Provider;
+}
+
 export interface Config {
   snapshot: SnapshotConfig;
   gnosisSafe: GnosisSafeConfig;
-  chainId: string;
+  zNA: zNAConfig;
+}
+
+export type zNA = string;
+export type zDAOId = string;
+
+export interface zDAO {
+  id: zDAOId; // Global zDAO identifier
+  zNA: zNA; // Linked zNA
+  // If not defined, title should be zNA by default
+  title?: string; // zDAO title
```

Comment:
If `title` is not defined and should be the `zNA` by default, then don't make `title` nullable and just return the `zNA` in it's place.